### PR TITLE
ref(ts): Convert IssueListTagFilter

### PR DESCRIPTION
**Pull request by davidenwang**
converted to typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/views/issueList/sidebar.tsx [hunk 1]
```diff
@@ -22,7 +22,6 @@ type DefaultProps = {
 };
 
 type Props = DefaultProps & {
-  orgId: string;
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/issueList/sidebar.tsx:25]: orgId is passed from overview -> sidebar -> tagfilter

but it seems like tagFilter doesn't even use orgId so it is now removed

File: src/sentry/static/sentry/app/views/issueList/tagFilter.tsx [hunk 2]
```diff
@@ -36,7 +48,7 @@ class IssueListTagFilter extends React.Component {
     this.api = new Client();
   }
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/issueList/tagFilter.tsx:49]: I think you can move all of this stuff out of the constructor

File: src/sentry/static/sentry/app/views/issueList/tagFilter.tsx [hunk 3]
```diff
@@ -1,32 +1,44 @@
...
+type State = {
+  query: string;
+  isLoading: boolean;
+  value: string | null;
+  textValue: string;
+  options?: SelectOption[];
+};
+
+class IssueListTagFilter extends React.Component<Props, State> {
+  static tagValueToSelectFormat = ({value}: TagValue): SelectOption => ({
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/issueList/tagFilter.tsx:33]: does this really have to be static on the class, can we just pull this function out?

Comment (davidenwang) [reply src/sentry/static/sentry/app/views/issueList/tagFilter.tsx:33]: good call, it seems like it's only used once for a `map` anyways so I'll just make it an anonymous function inside of there.

**Review by scttcper — APPROVED** [review]
looks good to me

**Review by matejminar — APPROVED** [review]

**Review by priscilawebdev — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Pull request merged**